src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -889,7 +910,7 @@ public static Cigar removeTrailingDeletions(final Cigar c) {
      * @param end Where should we stop keeping bases on the reference?  The maximum value is cigar.getReferenceLength()
      * @return a new Cigar with reference length == start - end + 1
      */
-    public static Cigar trimCigarByReference(final Cigar cigar, final int start, final int end) {
+    public static Triple<Cigar, Integer, Integer> trimCigarByReference(final Cigar cigar, final int start, final int end) {
```

Comment:
I would recommend making a static inner class to the `CigarBuilder` that is something the the effect of `CigarAndTrimmingInformaiton` that stores these things. Usually when we find ourselves using Triples its easier to make a helper class to document what all the fields are and prevent confusion.

Reply:
done